app/controllers/tracks/concepts_controller.rb

```diff
@@ -8,11 +8,20 @@ class Tracks::ConceptsController < ApplicationController
   def index
     @concept_map_data = Track::DetermineConceptMapLayout.(@track)
 
-    @concept_map_data[:status] = if current_user&.joined_track?(@track)
-                                   UserTrack::GenerateConceptStatusMapping.(@user_track)
-                                 else
-                                   {}
-                                 end
+    @concept_map_data[:status] =
+      if current_user&.joined_track?(@track)
+        UserTrack::GenerateConceptStatusMapping.(@user_track)
+      else
+        {}
+      end
+
+    @concept_map_data[:exercise_counts] =
+      if current_user&.joined_track?(@track)
+        UserTrack::GenerateConceptExerciseMapping.(@user_track)
+      else
+        {}
+      end
+
```

Comment:
```suggestion
      if current_user&.joined_track?(@track)
        @concept_map_data[:status]  = UserTrack::GenerateConceptStatusMapping.(@user_track)
        @concept_map_data[:exercise_counts] = UserTrack::GenerateConceptExerciseMapping.(@user_track)
      else
        @concept_map_data[:status] = {}
        @concept_map_data[:exercise_counts] = {}
      end
```

It would be nice if `UserTrack::GenerateConceptStatusMapping` and `UserTrack::GenerateConceptExerciseMapping` returned hashes if the user_track that is passed to them is nil, rather than us having to do this conditional dance here.

Reply:
done, will pr patch